octodns/idna.py

```diff
@@ -0,0 +1,32 @@
+#
+#
+#
+
+from idna import decode as _decode, encode as _encode
+
+
+def idna_encode(name):
+    # Based on https://github.com/psf/requests/pull/3695/files
+    # #diff-0debbb2447ce5debf2872cb0e17b18babe3566e9d9900739e8581b355bd513f7R39
+    try:
+        name.encode('ascii')
+        # No utf8 chars, just use as-is
+        return name
+    except UnicodeEncodeError:
+        if name.startswith('*'):
+            # idna.encode doesn't like the *
+            name = _encode(name[2:]).decode('utf-8')
+            return f'*.{name}'
+        return _encode(name).decode('utf-8')
+
+
+def idna_decode(name):
+    pieces = name.lower().split('.')
+    if any([p.startswith('xn--') for p in pieces]):
```

Comment:
Slightly better?

```suggestion
    if any(p.startswith('xn--') for p in pieces):
```

Reply:
:cool: guess I didn't know the list comprehension would work w/o the `[]`

Reply:
I think in my version, `any` will get an iterator which will stop evaluating subsequent elements after it finds a `True`.